tests/proc/tunderscore.nim

```diff
@@ -0,0 +1,45 @@
+discard """
+  targets: "c cpp js"
+"""
+
+block:
+  proc ok(_, _: int) = discard
+  ok(4, 2)
+
+block:
+  proc ok(_, _: int) = discard
+  ok(4, 2)
+
+block:
+  proc ok(_: int, _: float) = discard
+  ok(1, 2.0)
+
+block:
+  proc ok(_: int, _: float, _: string) = discard
+  ok(1, 2.6, "5")
+
+proc main() =
+  var x = 0
+
+  block:
+    proc foo(_, _: int) = x += 5
+
+    foo(1, 2)
+    doAssert x == 5
+
+  block:
+    proc foo(_: int, _: float) = x += 5
+
+    foo(1, 2)
+    doAssert x == 10
+
+  block:
+    proc foo(_: int, _: float, _: string) = x += 5
+
+    foo(1, 2, "5")
+    doAssert x == 15
+
```

Comment:
add an example with a template:
```suggestion
  block:
    template foo(_, _, a): int = a
    doAssert fo(1,2,3) == 3
```

(and IMO you can simplify the other examples using the same pattern as here)

Reply:
make sense